containers/podman-desktop
# feat: Fixed headers, improved scrollbars

### What does this PR do?

I have hit and tried to fix scrolling issues twice in the past and failed. This time I had a little help from Mairin and was a little more determined. This change makes all headers fixed (including titles and table headers) and moves all scrollbars to the 'content' area of the page.

Some notes:
- To keep headers & titles fixed and scroll the contents below, there were several places where padding needed to be moved 'up' or 'down' one level so that scrollbars are always on the correct/right edge.
- Likewise, once we have scrollbars in the correct places, it was clearer that (e.g.) several Settings pages had different top or bottom padding, had to fix these.
- I'm not entirely sure why the NavPage had separate table vs empty slots, but keeping them separate was causing layout issues. Most pages are no longer using both.

Intentionally left for future PRs:
- We can remove one of the NavPage `table` and `empty` slots and rename the other.
- The Settings > Preferences search should be part of the header now.
- In Settings > Resources > Create, the breadcrumb is in the header but the icon/title probably should be too?

### Screenshot/screencast of this PR

Before:
<img width="1162" alt="Screenshot 2023-06-13 at 1 39 26 PM" src="https://github.com/containers/podman-desktop/assets/19958075/a00a82a9-4b2b-4d7a-bb30-2f8ec0f7200d">

After:

https://github.com/containers/podman-desktop/assets/19958075/df37327d-8e9f-4743-abfb-ccbf384a7850

### What issues does this PR fix or reference?

Fixes #2864.
Fixes #2182.

### How to test this PR?

Open every screen, make window too big and too small and confirm content looks good. Stop all providers, do the same.
